capactio/capact
# Support backend context for typeinstance commands

## Description

Changes proposed in this pull request:
- support for modifying the `backend.context` by `capact ti edit`
- comment out the `value` property if the `acceptValue` in the backend storage is `false`
- do not display the `backend.context` if `contextSchema` is empty

## Testing

1) Follow the instruction from this [PR ](https://github.com/capactio/capact/pull/694), step 1-5. It will allow using `contextSchema` as an object.   
2) Get the download TI with the `--export` flag:  
`capact ti get $DOWNLOAD_TI --export -ojson`
```
{
  "id": "4c4c932e-b778-49cc-94a6-ea74a7d8f5a3",
  "typeInstance": {
    "attributes": [],
    "backend": {
      "context": {
        "provider": "dotenv"
      }
    },
    "value": {
      "key": "true"
    }
  }
}
```
3. View the edit mode with `capact ti edit $DOWNLOAD_TI`. The value should not be commented and the backend context is editable. 
4. Install the [Helm Storage](https://capact.io/docs/next/feature/storage-backends/helm/#installation) and [Mattermost](https://capact.io/docs/example/mattermost-installation/) on the cluster. As the `acceptValue` in the Helm Storage is `false` it is a good way to verify the functionality.
5. Verify that the TI which used `cap.type.helm.release.storage` has `value` commented out. 

There is an option to mock this behavior by rebuilding the client and modifying the `if backendData != nil && !backendData.AcceptValue` statement. 

## Related issue(s)
- https://github.com/capactio/capact/issues/689
